internal/nostr/nostr.go

```diff
@@ -297,20 +296,20 @@ func (svc *Service) NIP47Handler(c echo.Context) error {
 		})
 	}
 
-	svc.Logger.WithFields(logrus.Fields{
-		"eventId":      requestData.SignedEvent.ID,
-		"walletPubkey": requestData.WalletPubkey,
-		"relayUrl":     requestData.RelayUrl,
-		"webhookUrl":   requestData.WebhookUrl,
-	}).Info("Checking for duplicate request event")
-
-	subscription := Subscription{}
 	requestEvent := RequestEvent{}
 	findRequestResult := svc.db.Where("nostr_id = ?", requestData.SignedEvent.ID).Find(&requestEvent)
 
 	if findRequestResult.RowsAffected != 0 {
+		svc.Logger.WithFields(logrus.Fields{
```

Comment:
Why does it go on to check if there's a response event, and if not continue? shouldn't we only process requests once?

Reply:
Yeah I too think those checks there are unnecessary, will change to simply return `EVENT_ALREADY_PUBLISHED` (without the response event) if the requestEvent ID is in our database, else try publishing it